source/codegen/templates/service_helpers.mako

```diff
@@ -434,30 +450,38 @@ auto value = ${grpc_streaming_type.lower()}_message.value();
 <%def name="streaming_handle_in_direction_array(data_type, grpc_streaming_type, streaming_param)">
 <%
    is_coerced = service_helpers.is_input_array_that_needs_coercion(streaming_param)
+   streaming_param_name = common_helpers._camel_to_snake(streaming_param['name'])
+   c_element_type_that_needs_coercion = service_helpers.get_c_element_type_for_array_that_needs_coercion(streaming_param)
+   underlying_param_type = common_helpers.get_underlying_type_name(streaming_param["type"])
+
 %>\
 % if common_helpers.supports_standard_copy_conversion_routines(streaming_param):
     auto data_array = ${grpc_streaming_type.lower()}_message.value();
-    std::vector<${data_type}> array(data_array.begin(), data_array.end());
+    std::vector<${data_type}> ${streaming_param_name}(data_array.begin(), data_array.end());
```

Comment:
<${data_type}>can you replace this as well with the new values you are storing.
I think you can, and you can remove data_type from the helper parameter itself.

Reply:
Can't replace it with underlying_param_type, look at the other comment for futher details.